### PR TITLE
Wordle Assignment

### PR DESCRIPTION
**Pull request by AkashAdarsh**

### CODE REVIEW

**Comment by AkashAdarsh** [comment]
Please review my code,
Thanks.

File: statistics.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            lines = content
+            file_pointer.close()
+
+        with open('statistics.txt', 'w') as file_pointer:
+            file_pointer.writelines(lines)
+            file_pointer.close()
+
+
+    def increment_total_games_played(self):
+        with open('statistics.txt', 'r+') as file_pointer:
```
Comment (harshagr18) [inline statistics.py:20]: Using pointers to track the game and the file pointer is a very good idea. Will definitely consider it the next time I write file related code.

File: user_interface.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+def check_if_entered_word_is_valid(word):
+    return word.isalpha() and len(word) == 5
+
+
+def has_user_guessed_the_right_word(word, codeword):
+    return word == codeword
+
+
+def get_spot_name(spot_val):
```
Comment (harshagr18) [inline user_interface.py:14]: Returning None, or False would be easier to understand and the function easier to use and document.

File: wordle.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+from user_interface import *
```
Comment (harshagr18) [inline wordle.py:1]: Code needs more comments to make it easier to understand

File: wordle.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+from dictionary import Dictionary
+from statistics import *
```
Comment (harshagr18) [inline wordle.py:3]: Importing everything from a self written file with limited functions would not be the best, prefer writing function names to make it easier to understand

File: wordle_solver.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+guess = ""
+feedback = ""
+guess_list = []
+
+try:
```
Comment (harshagr18) [inline wordle_solver.py:5]: Really like the use of try-catch-exception, to make sure the file exists when any user is playing the game.

File: dictionary.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            self.set_word(word)
+        return self.get_word()
+
+    @staticmethod  # static method as does not return
+    def check_if_all_words_have_been_used():
+        valid_words_count = sum(1 for _ in open('valid_words.txt'))
+        used_words_count = sum(1 for _ in open('used_words.txt'))
+        if valid_words_count == used_words_count:
+            open('used_words.txt', 'w').close()
+
```
Comment (yashg772244) [inline dictionary.py:33]: This really was a good introduction and code looks really neat.

File: letterFrequency.csv [hunk 7]
```diff
@@ -0,0 +1 @@
+z, ,-,>, ,[,0,",", ,0,",", ,0,",", ,0,",", ,0,]
```
Comment (yashg772244) [inline letterFrequency.csv:1]: The letterFrequency.csv should not be the way it is. Kindly look into it.

File: logs.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+def log_to_file(level, message):
+    logging.basicConfig(filename='app.log', filemode='a+', format='%(levelname)s - %(message)s', level=logging.DEBUG)
+    if level == 'info':
+        logging.info(message)
+    elif level == 'error':
+        logging.error(message)
+    else:
+        logging.warning(message)
+
+
```
Comment (yashg772244) [inline logs.py:15]: This definitely looks good, but there could have been a better way to do it!!

File: statistics.py (hunk 1 above)
Comment (yashg772244) [reply statistics.py:20]: Totally agreed there. Such a great idea!!

File: user_interface.py (hunk 2 above)
Comment (yashg772244) [reply user_interface.py:14]: I see what you mean there. Definitely makes it easier.

File: user_interface.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            result.append('NA')
+            continue
+        if user_word_letter != code_arr[letter_index]:
+            result.append(False)
+            continue
+        result.append(True)
+
+    return result
+
+
```
Comment (yashg772244) [inline user_interface.py:40]: This was a great introduction to validate the letter position.

File: wordle.py [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+                continue
+
+            print('The word you entered -> %s' % user_word)
+            user_word = user_word.lower()
+
+            position_arr = position_check_for_letters(user_word, codeword)
+            validate_letter_position(position_arr, user_word)
+            trial += 1
+        self.post_gameplay()  # using class
+
```
Comment (yashg772244) [inline wordle.py:42]: The code looks good!! The post_gameplay looks good as well!! Definitely more comments would help.

File: wordle_solver.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+
+for guesses in range(6):
+    guess = input("\nword:").lower()
+    print("g - green, y - yellow, w - wrong / grey")
+    feedback = input("Feedback").lower()
+    if feedback == "ggggg":
+        print("Well Done! Guess", guesses + 1)
+        break
+
```
Comment (yashg772244) [inline wordle_solver.py:20]: This looks great!! Great job here!!!

**Review by yashg772244 — COMMENTED** [review]
Mentioned a few changes. Looks good overall!!!

**Comment by AkashAdarsh** [comment]
Taking all reviews into considerations, I have tried my best to make the appropriate changes and now merging to the branch.

**Pull request merged**